### ISSUE TITLE
Dosbox-X crash when changing keyboard layout 

### ISSUE THREAD

**Issue by JambonQuantique** [joncampbell123/dosbox-x#3487]
### Describe the bug

Hi there,

I'm setting up a Windows 98 Dosbox and, as a French, I need a French Layout.

Changing keyboard layout = fr or KEYB FR189 in auto exec result in crashing DosBox when Windows 98 boot up.

Bringing it back to "auto" seems ok, but none of numbers on my MacbookAir M1 are working under windows 98.

Hope it will be solved ! 

Thanks again a lot ! <3 

### Steps to reproduce the behaviour

1. Change or set kayboardlayout = fr in .conf file
2. Restart Dosbox-X with config file ... (tested with SDL 1 and 2)
3. Windows 98 will boot and then crash 

### Expected behavior

I exempt it to boot up correctly 

### What operating system(s) this bug have occurred on?

macOS Monterey 12.3.1

### What version(s) of DOSBox-X have this bug?

0.83.25 (SDL 1 and 2)

### Used configuration

```ini
Conf file is too long !!
```

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by rderooy** [comment]
I cannot test this right now, and this does sound like a bug, but if you're booting another OS the keyboard layout in DOSBox-X becomes irrelevant. The keyboard layout must be set by the OS you are booting, in this case Windows 98.

**Comment by JambonQuantique** [comment]
In fact, in Windows 98, when I set the keyboard layout on US, I've the correct AZERTY layout, and if I set it for French, I'm on QWERTY. Even if my conf file is set top fr or us.

And, the more problematic is that I haven't numbers working (keyboard without numeric pad) !

I continue my research!

Thanks a lot

**Comment by rderooy** [comment]
Are you using the SDL2 version of DOSBox-X? If not, please do so.

**Comment by JambonQuantique** [comment]
I'm using SDL1, as SDL2 crash on windows 98 start up...

[SDL2Crash.txt](https://github.com/joncampbell123/dosbox-x/files/8687171/SDL2Crash.txt)

**Comment by rderooy** [comment]
ahh an ARM Apple device. Cannot help you with that.

What I can tell you is that SDL1 was never meant to handle non-US keyboards. And even though DOSBox-X has some SDL1 hacks to improve matters, it is not perfect.

**Comment by JambonQuantique** [comment]
Ok, thanks anyway, If I find a kind of solution, I will report it here =)

**Comment by maron2000** [comment]
If you use SDL1 try `usescancodes=true`
The default is `auto`

**Comment by JambonQuantique** [comment]
Hello maron2000, 

I've give it a try, and on a ARM Mac, `usescancodes=true` crash dosbox on Windows 98 boot.

But, I've also tried on an intel Mac, and the keyboard works as excepted (like on your screenshot) so I guess this is an ARM issue...

**Comment by maron2000** [comment]
@JambonQuantique Too bad to hear that it didn't work.
Just to be sure, did you try with `keyboard layout = auto` since you mentioned you it didn't crash in such case in you primary comment.

**Comment by JambonQuantique** [comment]
Yep, I tested all the possible configurations, and it crashed... Thanks anyway for helping me! Hoping it will be more stable for ARM macs in the future ;)

**Comment by JambonQuantique** [comment]
Hi again,

I've finally found a solution : setting `core` to `normal` seems to stop windows 98 for crashing on boot. And then, I can use SDL2 and use the larboard layout of my choice :)
